Exercise_4.java

```diff
@@ -34,7 +34,27 @@ static void insert(Node temp, int key)
     { 
 
         // Do level order traversal until we find 
-        // an empty place and add the node.  
+        // an empty place and add the node. 
+        Queue<Node> nodeq = new LinkedList<Node>();
+        if(temp==null) temp = Node(key);

```

Comment:
initialization is incorrect, try to handle the syntax and minimize the errors as much as possible.